static/Lesson.js

```diff
@@ -147,29 +147,71 @@ const LessonShow = {
 }
 //URL.createObjectURL(new Blob(Uint8Array.from(s.content.slice(2).match(/../g),a=>parseInt(a,16)), { type: "image/jpeg" } ))
 const LessonList = {
-    props: ['tag'],
+    props: ['id'],
     template: `
-    <p v-if="!lessons.length">
-        No lessons. You can Import them with
+    <p v-if="lessons==null">
+        loading
+    </p>
+    <p v-else-if="lessons.length===0">
+        No lesson with categorie {{$props.id}}. You can Import them with
         <pre>make lesson_init</pre>
         then <a href=>refresh</a> this page.
     </p>
     <div class=grid>
-        <template v-for="(l,index) in lessons" >
-            <router-link :to="/lesson/+l.id" v-if="l.tags.includes(this.$props.tag)" class="card" style="border-radius: 1em">
-                <img alt=cover :src="l.icon||'/media/icons/kitchen.png'" style="padding: 15%;">
-                <span class="is-center">{{l.title}}</span>
+            <router-link v-for="lesson in lessons" :to="/lesson/+lesson.id" class="card" style="border-radius: 1em">
+                <img :src="lesson.icon||'/media/icons/kitchen.png'" style="padding: 15%;" alt="cover">
+                <span class="is-center">{{lesson.title}}</span>
             </router-link>
-        </template>
     </div>
     `
     ,
-    data() { return { lessons: []} },
-    async mounted() {
-        this.lessons = await (await fetch(`/api/lesson?select=id,title,owner,draft,tags,icon`)).json();
-    },
-
+    data() { return { lessons: null } },
+    watch: {
+        id: {
+            handler: async function (value) {
+                if(this.$props.id)
+                    this.lessons = await (await fetch(`/api/lesson?select=id,title,owner,draft,tags,icon&tags=cs.{group:${value}}`)).json();
+                else
+                    this.lessons = await (await fetch(`/api/lesson?select=id,title,owner,draft,tags,icon&tags=cs.{type:vocab}`)).json();
+            },
+            immediate: true
+        }
+    }
+}
 
+const CategoriesList = {
+    //props: ['tag'],
+    template: `
+    <div class=grid>
+        <router-link  v-for="category in categories" :to="/category/+category" class="card" style="border-radius: 1em">
+            <img :src='"/media/icons/"+category+".svg"' style="padding: 15%;" alt="cover">
+            <span class="is-center text-capitalize">{{category}}</span>
+        </router-link >
+    </div>
+    `
+    ,
+    data() { return { categories: ["vocab","conversation","grammar","reading","listening","game","test"] } },
```

Comment:
bad practice,
shall be fetch from database + grouped , see:
- `fetch(/lesson?select=tags)`
- [Set()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) for groupping